### PR TITLE
fix(secu): prevent from sql injections from common xml model

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

- avoid SQL injection from unused common XML template by removing the file and the folder

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [x] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

please contact me

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 1]
```diff
@@ -48,47 +47,46 @@
...
+    array('options' => array('default' => $obj->defaultCriticality))
+);
+//if instance value is not set, displaying all active pollers linked resources
+$instance = filter_var($obj->defaultPoller ?? -1, FILTER_VALIDATE_INT);
+$hostgroups = filter_var($obj->defaultHostgroups ?? 0, FILTER_VALIDATE_INT);
+
+$search = filter_input(INPUT_GET, 'search', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+$statusHost = filter_input(INPUT_GET, 'statusHost', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+$statusFilter = filter_input(INPUT_GET, 'statusFilter', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+$order = filter_input(
```
Comment (callapa) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]: Please stop using regex to simply test a value with two possible cases.
Do not use a mass to drive a nail.

File: www/include/monitoring/status/Services/xml/serviceGridXML.php [hunk 2]
```diff
@@ -35,55 +35,52 @@
...
+$instance = filter_var($obj->defaultPoller ?? -1, FILTER_VALIDATE_INT);
+$hostgroups = filter_var($obj->defaultHostgroups ?? 0, FILTER_VALIDATE_INT);
+$search = filter_input(INPUT_GET, 'search', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+$sort_type = filter_input(
+    INPUT_GET,
+    'sort_type',
+    FILTER_SANITIZE_STRING,
+    array('options' => array('default' => 'host_name'))
+);
+$order = filter_input(
```
Comment (callapa) [inline www/include/monitoring/status/Services/xml/serviceGridXML.php]: Please stop using regexes to simply test a value with two possible cases.
Do not use a mass to drive a nail

File: www/include/monitoring/status/Services/xml/serviceSummaryXML.php [hunk 3]
```diff
@@ -35,61 +35,59 @@
...
+$instance = filter_var($obj->defaultPoller ?? -1, FILTER_VALIDATE_INT);
+$hostgroups = filter_var($obj->defaultHostgroups ?? 0, FILTER_VALIDATE_INT);
+$search = filter_input(INPUT_GET, 'search', FILTER_SANITIZE_STRING, array('options' => array('default' => '')));
+$sort_type = filter_input(
+    INPUT_GET,
+    'sort_type',
+    FILTER_SANITIZE_STRING,
+    array('options' => array('default' => 'host_name'))
+);
+$order = filter_input(
```
Comment (callapa) [inline www/include/monitoring/status/Services/xml/serviceSummaryXML.php]: Please stop using regexes to simply test a value with two possible cases.
Do not use a mass to drive a nail

File: www/include/monitoring/status/Services/xml/serviceXML.php [hunk 4]
```diff
@@ -84,85 +70,80 @@
...
+    FILTER_SANITIZE_STRING,
+    array('options' => array('default' => ''))
+);
+$sort_type = filter_input(
+    INPUT_GET,
+    'sort_type',
+    FILTER_SANITIZE_STRING,
+    array('options' => array('default' => 'host_name'))
+);
+$order = filter_input(
```
Comment (callapa) [inline www/include/monitoring/status/Services/xml/serviceXML.php]: Please stop using regexes to simply test a value with two possible cases.
Do not use a mass to drive a nail

Comment (sc979) [reply www/include/monitoring/status/Services/xml/serviceXML.php]: I'll replace these by whitelist ASAP ;)

**Pull request merged**